downloader/src/main/kotlin/vcs/Git.kt

```diff
@@ -59,30 +58,32 @@ const val GIT_HISTORY_DEPTH = 50
 class Git : VersionControlSystem(), CommandLineTool {
     companion object {
         init {
-            // Configure JGit to connect to the SSH-Agent if it is available.
             val sessionFactory = object : JschConfigSessionFactory() {
-                override fun configure(hc: OpenSshConfig.Host, session: Session) {
-                    session.setConfig("StrictHostKeyChecking", "no")
-                }
+                @Suppress("EmptyFunctionBlock")
+                override fun configure(hc: OpenSshConfig.Host, session: Session) {}
+
+                override fun configureJSch(jsch: JSch) {
+                    // The default JSch instance uses identity files named "identity", "id_rsa" or "id_dsa" from the
+                    // current user's ".ssh" directory, see https://www.codeaffine.com/2014/12/09/jgit-authentication/.
 
-                override fun createDefaultJSch(fs: FS): JSch {
-                    val jSch = super.createDefaultJSch(fs)
+                    // Accept unknown hosts.
+                    JSch.setConfig("StrictHostKeyChecking", "no")
+
+                    // Limit to "publickey" to avoid "keyboard-interactive" prompts.
+                    JSch.setConfig("PreferredAuthentications", "publickey")
```

Comment:
If the above two settings are global they could be moved out of the session factory to the top of the `init` block.

Reply:
Correct. I decided to still keep them here as also [JGit itself](https://github.com/eclipse/jgit/blob/231c44d553d9868795eafd9c8d87a2d3f7de50bd/org.eclipse.jgit/src/org/eclipse/jgit/transport/JschConfigSessionFactory.java#L388-L396) calls `JSch.setConfig()` in `createDefaultJSch()`, not in `configureJSch()`.